packages/jest-cli/src/cli/runCLI.js

```diff
@@ -45,6 +45,17 @@ module.exports = (
     return;
   }
 
+  const config = readConfig(argv, root);
+  if (argv.showConfig) {
+    const output = ({config}) => {
+      pipe.write(JSON.stringify(config, null, 2));
```

Comment:
I said back in the issue that it appeared to be 2 changes here. One being adding this flag and outputting JSON config. The other being this change, to change `--debug` output to be JSON as well. I'm willing to make the change in this PR, but I figured they would be separate changes therefor separate PRs.